devito/finite_differences/differentiable.py

```diff
@@ -8,7 +8,8 @@
 from cached_property import cached_property
 from devito.finite_differences.lazy import Evaluable
 from devito.logger import warning
-from devito.tools import EnrichedTuple, filter_ordered, flatten
+from devito.tools import filter_ordered, flatten
+from devito.types.utils import DimensionTuple
 
 __all__ = ['Differentiable']
 
```

Comment:
I don't know where to write this, so I'll write it here.

Shouldn't `_fd_priority` be defined in `Differentiable` so that any future subclasses have it defined (even though with a dummy value). That is

```
class Differentiable(...):

    _fd_priority = some dummy value
     """thourhgout explanation of what this represents."""
```

Reply:
It is defined in Differentiable below, just different value whether time dependent or not

Reply:
ah right sorry.